pcsx2/IopBios.cpp

```diff
@@ -100,632 +102,898 @@ namespace R3000A {
 #define Ra2 (iopMemReadString(a2))
 #define Ra3 (iopMemReadString(a3))
 
-static std::string host_path(const std::string path)
-{
-// WIP code. Works well on win32, not so sure on unixes
-// TODO: get rid of dependency on CWD/PWD
-#if USE_HOST_REWRITE
-	// we want filenames to be relative to pcs2dir / host
-
-	std::string pathMod;
+#define FIO_SO_IXOTH 0x0001
+#define FIO_SO_IWOTH 0x0002
+#define FIO_SO_IROTH 0x0004
 
-	// partial "rooting",
-	// it will NOT avoid a path like "../../x" from escaping the pcsx2 folder!
+#define FIO_SO_IFLNK 0x0008
+#define FIO_SO_IFREG 0x0010
+#define FIO_SO_IFDIR 0x0020
 
-
-	const std::string _local_root = "/usr/local/";
-	if (HOST_REWRITE_USR_LOCAL && 0 == path.compare(0, _local_root.size(), _local_root.data())) {
-		return HostRoot + path.substr(_local_root.size());
-	} else if ((path[0] == '/') || (path[0] == '\\') || (isalpha(path[0]) && (path[1] == ':'))) // absolute NATIVE path (X:\blah)
+	static std::string host_path(const std::string path)
 	{
-		// TODO: allow some way to use native paths in non-windows platforms
-		// maybe hack it so linux prefixes the path with "X:"? ;P
-		// or have all platforms use a common prefix for native paths
-		// FIXME: Why the hell would we allow this?
-		return path;
-	} else // relative paths
-		return HostRoot + path;
-
-	return pathMod;
-#else
-	return path;
-#endif
-}
-
-// TODO: sandbox option, other permissions
-class HostFile : public IOManFile
-{
-public:
-	int fd;
+		// We are NOT allowing to use the root of the host unit.
+		// For now it just supports relative folders from the location of the elf
+		if (path.rfind(HostRoot, 0) == 0)
+			return path;
+		else // relative paths
+			return HostRoot + path;
+	}
 
-	HostFile(int hostfd)
+	static int host_stat(const std::string path, fio_stat_t* host_stats)
 	{
-		fd = hostfd;
-	}
+		struct stat file_stats;
 
-	virtual ~HostFile() = default;
+		if (::stat(path.c_str(), &file_stats))
+			return -IOP_ENOENT;
 
-	static __fi int translate_error(int err)
-	{
-		if (err >= 0)
-			return err;
+		host_stats->size = file_stats.st_size;
+		host_stats->hisize = 0;
 
-		switch(err)
+		// Convert the mode.
+		host_stats->mode = (file_stats.st_mode & (FIO_SO_IROTH | FIO_SO_IWOTH | FIO_SO_IXOTH));
+#ifndef _WIN32
+		if (S_ISLNK(file_stats.st_mode))
 		{
-			case -ENOENT:
-				return -IOP_ENOENT;
-			case -EACCES:
-				return -IOP_EACCES;
-			case -EISDIR:
-				return -IOP_EISDIR;
-			case -EIO:
-			default:
-				return -IOP_EIO;
+			host_stats->mode |= FIO_SO_IFLNK;
 		}
+#endif
+		if (S_ISREG(file_stats.st_mode))
```

Comment:
Think you need to include sys/stat.h for it to work on windows, don't know if underscore is needed or not.

Reply:
Fixed! Now it also compiles in Windows